src/utils/Snippets.ts

```diff
@@ -0,0 +1,101 @@
+import {OpenAPIV3} from "openapi-types";
+import { mock } from 'mock-json-schema';
+
+import { DeRefResponse, deRef, recursiveDeRef } from './Openapi';
+import {Request as RequestFormat, Header, QueryString, PostData} from 'har-format'
+
+
+const getHeaders = (params: DeRefResponse<OpenAPIV3.ParameterObject>[], document: OpenAPIV3.Document): Header[] => {
+  const headers: Header[] = []
+
+  if (document.components?.securitySchemes) {
+    Object.values(document.components?.securitySchemes).forEach(s => {
+      const scheme = deRef(s, document)
+      if ("in" in scheme && scheme.in === "header") {
+        headers.push({name: scheme.name, value: scheme.type})
+      }
+    });
+  }
+  params.forEach(param => {
+    if (param.in === "header" && param.required) {
+      let val = param.name
+      if (param.schema && "type" in param.schema) {
+        val = param.schema.type as string
+      }
+      headers.push({name: param.name, value: val})
+    }
+  })
+
+  // setting 'Content-Type' 'application/json' as a default header
+  const hasContentType = headers.some(header => header.name === 'Content-Type');
+  if (!hasContentType) {
+    headers.push({ name: 'Content-Type', value: 'application/json' });
```

Comment:
I think that we should honor whatever the API has in the content instead of the headers:

![image](https://user-images.githubusercontent.com/3845764/229814555-7a63765e-70bd-45c0-8df2-40bdc7c6f13a.png)

Reply:
I wrote a function to infer the content type. It goes like this:
* If content type is mentioned in the `requestBody`, use that
* Else if content type is mentioned in the `responses`, use that
* use `application/json` as default otherwise